### DISCUSSION TITLE
PowerDNS Subdomain add DNS Records Problem Error: This is not a valid CNAME. There already exists a record with this name.

### DISCUSSION THREAD

**Discussion by verifyersin** [PowerDNS/pdns#13487] [Q&A] [answered]
- [ ] This is not a support question, I have read [about opensource](https://www.powerdns.com/opensource.html) and will send support questions to the IRC channel, [Github Discussions](https://github.com/PowerDNS/pdns/discussions/) or the mailing list.
- [ ] I have read and understood the ['out in the open' support policy](https://blog.powerdns.com/2016/01/18/open-source-support-out-in-the-open/)

 - Program: Authoritative, Recursor, dnsdist 
 - Issue type: Bug report

### Short description

We use Powerdns. However, we have a problem as follows. 

When we want to enter a CNAME record for a subdomain, it gives a warning as follows.

Error: This is not a valid CNAME. There already exists a record with this name.

To give an example, the existing records are;

subdomain.example.com MX mail.xxx.com
subdomain.example.com TXT "v=spf1 include:xxxx.com -all"

The CNAME record to be added is as follows.

subdomain.example.com CNAME xxxx.com

### Environment

 - Operating system: AlmaLinux release 8.6 (Sky Tiger)
 - Software version:  pdnsutil 4.7.0-alpha1
 - Software source: poweradmin v2.2.2

### Steps to reproduce

1. One MX record for the subdomain,
2. Enter the TXT record for the subdomain.
3. You will receive the error when you want to enter the Cname record for the subdomain.

**Comment by rgacogne** [comment, marked as answer]
It's not possible to have a CNAME and other types for the same name, see https://datatracker.ietf.org/doc/html/rfc1912#section-2.4

**Comment by verifyersin** [comment]
I think I didn't convey my problem correctly.

I have 2 records specified in the file named Screenshot_5.jpg.

When I want to add a CNAME record for the address e.example.com

I'm getting the error mentioned in Screenshot_4.jpg.

Thank you for your help,
![Screenshot_5](https://github.com/PowerDNS/pdns/assets/116286107/083ac530-e79f-45ef-99fb-032470413c0e)
![Screenshot_4](https://github.com/PowerDNS/pdns/assets/116286107/79b7fb09-8b7a-40ae-aa3e-862d6482179b)

**Comment by rgacogne** [comment]
This is not possible, the DNS protocol does not allow that. If you want to have a CNAME for `e.example.com` it needs to be the only type for that name, so you will have to remove the MX and TXT types first.
